docs/src/main/asciidoc/optaplanner.adoc

```diff
@@ -0,0 +1,1034 @@
+
+In Maven, your `pom.xml` file has the following content:
+
+[source,xml]
+----
+<?xml version="1.0"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.acme</groupId>
+  <artifactId>acme-optaplanner</artifactId>
+  <version>1.0.0-SNAPSHOT</version>
+
+  <properties>
+    <quarkus.version>...</quarkus.version>
+    <compiler-plugin.version>3.8.1</compiler-plugin.version>
+    <surefire-plugin.version>2.22.1</surefire-plugin.version>
+    <maven.compiler.target>1.8</maven.compiler.target>
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+  </properties>
+
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-bom</artifactId>
+        <version>${quarkus.version}</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+  <dependencies>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-optaplanner</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-optaplanner-jackson</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-resteasy</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-resteasy-jackson</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-junit5</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>io.rest-assured</groupId>
+      <artifactId>rest-assured</artifactId>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>${compiler-plugin.version}</version>
+      </plugin>
+      <plugin>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-maven-plugin</artifactId>
+        <version>${quarkus.version}</version>
+        <executions>
+          <execution>
+            <goals>
+              <goal>build</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <version>${surefire-plugin.version}</version>
+        <configuration>
+          <systemProperties>
+            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+          </systemProperties>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+</project>
+----
```

Comment:
Could you please follow how we do it in the other guides? We usually point to the quickstart and provide the command line to create the project. We don't put the Maven file.

Reply:
- The lack of the maven file makes it harder to build the example as a user.
- Also, in the first step, the maven file doesn't add in hibernate. Only at the end of the guide we talk about adding hibernate and adjusting the SolverManager's usage pattern accordingly. But the intermediate solution still makes sense for many use cases too.

Therefor, I strongly recommend keeping the maven file. Wydt?

Reply:
You don't need the pom file if you are following the same pattern as the other guides: https://github.com/quarkusio/quarkus/blob/master/docs/src/main/asciidoc/hibernate-search-elasticsearch.adoc#creating-the-maven-project .

The command will scaffold the project for you properly.

I don't understand why it wouldn't work in your case?

Reply:
Happy to do that to get this merged - but I think it's naive to presume that most people that follow the guide start from scratch: some will do it on top of an existing project (I always did it that way). The spring guides show the entire maven and gradle file, probably for that reason.

It's not uncommon for users reading the guide to take their existing project (often result of another guide), copy paste the missing dependencies on it and start doing a similar thing as the guide, only to refactor it - once it works - into what they exactly need. At least I do it like that more often than not to acquire a new technology :)

Reply:
@gsmet I 'll reduce the pom.xml to just the dependencies and refer for the rest like the other quarkus guides. Ok?

Reply:
Done, also added some info around Gradle and added the guide url in optaplanner's quarkus-extension.yaml

Reply:
I think I would remove the Gradle instructions to be in line with what we do in the other guides. People can translate easily.

Reply:
They can translate, but they want to be able to copy-paste (most don't start from scratch). If Gradle is a first class citizen of Gradle, and we want Quarkus to feel friendly for Gradle users, then these need to be here too, just like in the Spring guides. Basically a "Maven first, Gradle too" story (just like a "Java first, Kotlin and Scala too" story).

I understand that for consistency you might not want some of the guides to have it and others not. I'd rather just keep them and create an issue to adjust the other guides too. Alternatively, we could comment out the gradle instructions for now... Wdyt?

![image](https://user-images.githubusercontent.com/176880/76291607-bdb76b00-62ad-11ea-9439-7158e6011e94.png)
Source: https://snyk.io/blog/jvm-ecosystem-report-2020/

Reply:
I think I would rather hide them for now.

I'm not sure we want to clutter each guide with Gradle stuff so we would have to discuss that first.

Reply:
As you wish

Reply:
@gsmet Done (hidden gradle instructions). Rebased on master and squashed. No more open conversations that I am aware of. Should be good to merge :)